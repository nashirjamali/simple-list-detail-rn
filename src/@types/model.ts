import {TRANSACTION_STATUS} from '../constants';

export type TransactionStatus =
  (typeof TRANSACTION_STATUS)[keyof typeof TRANSACTION_STATUS];

export type Transaction = {
  id: string;
  amount: number;
  unique_code: number;
  status: TransactionStatus;
  sender_bank: string;
  account_number: string;
  beneficiary_name: string;
  beneficiary_bank: string;
  remark: string;
  created_at: string;
  completed_at: string;
  fee: number;
};

export type Transactions = {
  [key: string]: Transaction;
};
