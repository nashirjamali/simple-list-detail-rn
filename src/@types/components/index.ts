export * from './GeneralText.types';
export * from './Button.types';
export * from './Layout.types';
export * from './TextInput.types';
export * from './Label.types';
export * from './Card.types';
