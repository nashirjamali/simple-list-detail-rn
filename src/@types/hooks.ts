import {Transaction, Transactions} from './model';

export type UseFetchReturns<TData> = {
  data: TData | null;
  loading: boolean;
  error: string | null;
};

export type TransactionContextType = {
  transaction: Transaction | null;
  setTransaction: (transaction: Transaction | null) => void;
};
