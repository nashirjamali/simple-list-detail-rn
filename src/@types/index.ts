export * from './components';
export * from './router';
export * from './model';
export * from './hooks';
export * from './screens';
