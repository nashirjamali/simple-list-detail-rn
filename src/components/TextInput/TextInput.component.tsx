import React, {FC} from 'react';
import {TextInput as TextInputRN, View} from 'react-native';

import type {ButtonProps, TextInputProps} from '../../@types';
import Button from '../Button/Button.components';
import styles from './TextInput.styles';

/**
 * A customizable text input component that includes a button for additional actions.
 *
 * @component
 * @param {TextInputProps} props - The properties for the TextInput component.
 * @param {string} props.placeholder - The placeholder text to display when the input is empty.
 * @param {string} props.value - The current value of the text input.
 * @param {function} props.onChangeText - Callback function that is called when the text input value changes.
 * @param {ButtonProps} [props.buttonComponentProps] - Additional props to pass to the button component.
 * @returns {JSX.Element} The rendered TextInput component with an associated button.
 */
const TextInput: FC<TextInputProps> = ({
  placeholder,
  value,
  onChangeText,
  buttonComponentProps,
  ...props
}: TextInputProps): JSX.Element => (
  <View style={styles.container}>
    <TextInputRN
      style={styles.input}
      placeholder={placeholder}
      onChangeText={onChangeText}
      value={value}
      {...props}
    />
    {buttonComponentProps && <Button {...buttonComponentProps} />}
  </View>
);

export default TextInput;
