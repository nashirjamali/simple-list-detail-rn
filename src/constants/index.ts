export * from './components';
export * from './Colors.constants';
