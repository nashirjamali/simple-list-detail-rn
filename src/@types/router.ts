import {ROUTES} from '../constants';

export type RootStackParamList = {
  [ROUTES.TRANSACTION_LIST]: undefined;
};
