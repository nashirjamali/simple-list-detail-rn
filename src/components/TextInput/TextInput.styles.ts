import {StyleSheet} from 'react-native';

import type {TextInputStyles} from '../../@types';
import {COLORS} from '../../constants';

const styles: TextInputStyles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.WHITE,
    borderRadius: 10,
    paddingRight: 10,
    paddingTop: 5,
    paddingBottom: 5,
    paddingLeft: 5,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    gap: 5,
    width: '100%',
  },
  input: {},
});

export default styles;
