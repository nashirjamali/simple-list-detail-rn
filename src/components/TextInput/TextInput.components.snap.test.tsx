import React from 'react';
import {render} from '@testing-library/react-native';

import {ButtonProps} from '../../@types';
import {BUTTON_VARIANTS} from '../../constants';

import TextInput from './TextInput.component';

describe('TextInput Component', () => {
  it('renders correctly with placeholder and value', () => {
    const {toJSON} = render(
      <TextInput
        placeholder="Enter text"
        value="Sample text"
        onChangeText={() => {}}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with button', () => {
    const buttonProps: ButtonProps = {
      title: 'Submit',
      variant: BUTTON_VARIANTS.NORMAL,
      onPress: jest.fn(),
    };

    const {toJSON} = render(
      <TextInput
        placeholder="Enter text"
        value="Sample text"
        onChangeText={() => {}}
        buttonComponentProps={buttonProps}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
