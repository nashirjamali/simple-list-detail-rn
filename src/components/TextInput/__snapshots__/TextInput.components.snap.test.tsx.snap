// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component renders correctly with button 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "row",
      "gap": 5,
      "justifyContent": "space-between",
      "paddingBottom": 5,
      "paddingLeft": 5,
      "paddingRight": 10,
      "paddingTop": 5,
      "width": "100%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter text"
    style={{}}
    value="Sample text"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "gap": 2,
        "justifyContent": "space-between",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FB6544",
          "fontSize": 14,
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;

exports[`TextInput Component renders correctly with placeholder and value 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "row",
      "gap": 5,
      "justifyContent": "space-between",
      "paddingBottom": 5,
      "paddingLeft": 5,
      "paddingRight": 10,
      "paddingTop": 5,
      "width": "100%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter text"
    style={{}}
    value="Sample text"
  />
</View>
`;
